File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    for i in range(n):
        if not visited[i]:
            visited[i] = True
            dfs(i, computers, visited)
            answer += 1
    
    return answer

def dfs(n, computers, visited):
    for i in range(len(computers[n])):
        if computers[n][i] == 0:
            continue
        if visited[i]:
            continue
        if computers[n][i] == 1 and not visited[i]:
            visited[i] = True
            dfs(i, computers, visited)
    
        
            

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visited;
    static int[][] arr;
    static int len;
    
    public void bfs(int f){
        Queue<Integer> queue = new LinkedList<>();
        visited[f] = true;
        queue.offer(f);
        while(!queue.isEmpty()){
            int val = queue.poll();
            for(int i=0; i<len; i++){
                if(!visited[i] && arr[val][i] == 1){
                    visited[i] = true;
                    queue.offer(i);
                }
            }
        }
    }

    public int solution(int n, int[][] computers) {
        int answer = 0;
        len = n;
        arr = computers;
        visited = new boolean[n];
        
        for(int i=0; i<n; i++){
            if(!visited[i]){
                bfs(i);
                answer++;
            }
        }
        
        return answer;
    }
}